datalad/support/gitrepo.py

```diff
@@ -937,14 +937,18 @@ def clone(cls, url, path, *args, **kwargs):
         # make sure that Git doesn't mangle relative path specification into
         # mildly obscure absolute paths
         # https://github.com/datalad/datalad/issues/3538
+        # Note, that POSIX is required even on windows, since it's an URL!
         if isinstance(url_ri, PathRI):
             path = Path(url)
             if not path.is_absolute():
-                # always in POSIX even on windows
-                path = path.as_posix()
-                if not path.startswith(op.pardir + '/'):
-                    path = posixpath.join(op.curdir, path)
-                gr.config.set('remote.origin.url', path,
+                # get git-created path
+                git_url = gr.config.get('remote.origin.url')
+                # Note: Not sure, whether there are circumstances where this is relative already
+                if op.isabs(git_url):
+                    # ... and make it a relative one
+                    # Note: Using posixpath here, since pathlib's relative_to isn't what you'd expect
+                    git_url = posixpath.relpath(git_url, gr.pathobj.as_posix())
+                gr.config.set('remote.origin.url', git_url,
```

Comment:
If the `op.isabs(git_url)` condition is false, the url hasn't changed from the configured one, so we might as well avoid writing it out again with `.set` (i.e. move this call under the condition).

Reply:
True.